scripts/azureml-assets/azureml/assets/release/model_publish_utils/model_publish_utils.py

```diff
@@ -0,0 +1,116 @@
+import os
+import shutil
+import yaml
+import logging
+from typing import List
+from azureml.assets.util import logger
+import azureml.evaluate.mlflow as mlflow
+from mlflow.models import ModelSignature
+from sqlalchemy import true
+from transformers import  AutoTokenizer, AutoConfig, pipeline
+from transformers import (
+    AutoModelForSequenceClassification,
+    AutoModelForMaskedLM,
+    AutoModelForTokenClassification,
+    AutoModelForQuestionAnswering,
+    AutoModelWithLMHead
+)
+
+class ModelPublishUtils:
+
+    @classmethod
+    def __init__(self, spec_path: str) -> None:
+        
+        self.model_yaml_path = spec_path + "/model.yml"
+        self.asset_yaml_path = spec_path + "/asset.yml"
+        self.model_name = self._get_model_name()
+        self.model_uri = self._get_model_uri()
+        self.model_commit = self._get_model_commit()
+        self.model_task_name = self._get_model_task_name()
+        self.model_dir_name = spec_path + self.model_name
+
+    @classmethod
+    def _get_model_name(self) -> str:
+        with open(self.model_yaml_path) as f:
+            model_file = yaml.safe_load(f)
+        return model_file["name"]
+
+    @classmethod
+    def _get_model_uri(self) -> str:
+        with open(self.asset_yaml_path) as f:
+            asset_file = yaml.safe_load(f)
+        return asset_file["package"]["path"]
+
+    @classmethod
+    def _get_model_commit(self) -> str/None:
+        with open(self.asset_yaml_path) as f:
+            asset_file = yaml.safe_load(f)
+        return asset_file["package"]["sha"] if asset_file["package"]["sha"] else None
+    
+    @classmethod
+    def _get_model_task_name(self) -> str:
+        with open(self.asset_yaml_path) as f:
+            asset_file = yaml.safe_load(f)
+        return asset_file["task"]
+
+    @classmethod
+    def _git_clone_model(self) -> None:
+        if self.validate_model_and_task():
+            cmd = f'git clone {self.model_uri} {self.model_dir_name}'
+            os.system(cmd)
+            if self.model_commit:
+                os.chdir(f'{self.model_dir_name}')
+                os.system(f'git reset --hard {self.model_commit}')
+
+    # Convert Model to ISV FLavoured MLFlow model and save it.
+    @classmethod
+    def _save_mlflow_model(self):
```

Comment:
@adurbhaka

Reply:
During our last conversation with Shipra and team, we got to know that pushing packages, metrics, evaluate-flow to PyPi would take a lot of effort and would be targeted only after private preview. So, till they are pushed to public, we have decided to use the packages from storage. We will be updating them as and when we recieve new versions.

Reply:
@skasturi : Haritha's point is correct. We are in process of publishing azureml-metrics as py-pi (with internal consumption only but that would be post PrP. We arent planning to publish azureml-evaluate-mlflow to py-pi rite now since the changes within are strictly internal.

Please let us know if you see this as blocker.

Reply:
Thanks @haritha-bendapudi , @ShipraJain01. 

I don't consider this a blocker. I am thinking that we should separate out the public and non-public stuff and create separate PRs for them. 

Model configuration etc can be part of this PR. HF Flavor in a separate PR & we keep iterating based on dev indices till the package pushed to PyPi. 

@wayliums , @elesel

Reply:
Made changes to this PR to make it model publish without HFTransformers flavor.